src/axom/spin/internal/linear_bvh/TraversalPredicates.hpp

```diff
@@ -127,6 +172,54 @@ class TraversalPredicates< DIMENSION_2, FloatType >
     return in_right;
   }
 
+  template < typename RayType >
+  AXOM_HOST_DEVICE
+  static inline bool rayIntersectsLeftBin(
+      const RayType& r, const vec4_t& s1, const vec4_t& s2,
+      FloatType TOL=numerics::floating_point_limits<FloatType>::epsilon()
+      ) noexcept
+  {
+    const FloatType& x0 = r[ 0 ];
+    const FloatType& y0 = r[ 1 ];
+    const FloatType& nx = r[ 2 ];
+    const FloatType& ny = r[ 3 ];
+
+    // extract left bin, see BVHData.hpp for the internal BVH layout
+    const FloatType& xmin = s1[ 0 ];
```

Comment:
Perhaps a utility function in ``BVHData.hpp`` could be used to hide the implementation details?

Reply:
Yes, I have been thinking about something like that. I'll mull it over some more and add that in a future PR.